src/wazuh_modules/wm_syscollector.c

```diff
@@ -32,6 +32,7 @@ pthread_mutex_t sys_stop_mutex;
 bool need_shutdown_wait = false;
 pthread_mutex_t sys_reconnect_mutex;
 bool shutdown_process_started = false;
+bool syscollector_initialized = false;
```

Comment:
Have you tried instead to declare the condition variable with a initial value like this?

```
pthread_cond_t sys_stop_condition = PTHREAD_COND_INITIALIZER;
```

It seems to prevent the crash also.

Reply:
I saw this and didn't consider this as a fix. Thanks, seems to be the way to go